packages/block-editor/src/components/inserter/hooks/use-patterns-state.js

```diff
@@ -18,17 +18,35 @@ import { __, sprintf } from '@wordpress/i18n';
  *
  * @return {Array} Returns the patterns state. (patterns, categories, onSelect handler)
  */
-const usePatternsState = ( onInsert ) => {
-	const { patternCategories, patterns } = useSelect( ( select ) => {
-		const {
-			__experimentalBlockPatterns,
-			__experimentalBlockPatternCategories,
-		} = select( 'core/block-editor' ).getSettings();
-		return {
-			patterns: __experimentalBlockPatterns,
-			patternCategories: __experimentalBlockPatternCategories,
-		};
-	}, [] );
+const usePatternsState = ( onInsert, selectedCategory ) => {
```

Comment:
Adding the `selectedCategory` to this hook change its semantics a bit.
I'd personally prefer to leave the hook as is and adapt the component behavior instead. Any particular reason that lead to this option instead?

Reply:
This is what I preferred and how I had it originally.

I wasn't able to make it work though, I could not pass a render time filtered array of category patterns as dependency of the `orderedPatterns` useMemo call without causing an infinite loop. It needed to be fetched from the inserter state for the lazy loading to work correctly. Someone much smarter than me might be able to figure out where I was going wrong. /cc @glendaviesnz @aaroncampbell

Reply:
@youknowriad reverted the hook and moved the filtering to the component, turned out the loop was because the asyncList method needs the same array reference handing to it each time, so fixed this by keeping the filtered list in component state until category changes ... is this more in line with what you were expecting?

Reply:
Thanks for working that through @glendaviesnz 💪